app/namespace.go

```diff
@@ -184,6 +200,29 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
 				return err
 			},
 			Subcommands: []*cli.Command{
+				{
+					Name:    "delete",
+					Usage:   "Delete a temporal namespace",
+					Aliases: []string{"d"},
+					Flags: []cli.Flag{
+						RequestIDFlag,
+						ResourceVersionFlag,
+						&cli.StringFlag{
```

Comment:
Should we ask the user to confirm deletion by typing in namespace name (and then I believe we have an auto-confirm flag if user wants to bypass it)?